### PR TITLE
Fixes #14068 - Allow creating content views with repository names

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

**Comment by ehelms** [comment]
Please squash into a single commit.

**Comment by komidore64** [comment]
i'm not sure what rubocop is doing...

**Comment by bbuckingham** [comment]
@adamruzicka, If possible, please go ahead and rebase with master.  I believe the following PR recently merged will address the rubocop failure: https://github.com/Katello/hammer-cli-katello/pull/361 .  Thanks!

**Comment by adamruzicka** [comment]
@komidore64, @bbuckingham: Thanks for looking into this. I did a rebase, let's hope it gets fixed

**Comment by bbuckingham** [comment]
@adamruzicka, Thanks! The fix tested out well for me.  ACK.

@ehelms, @komidore64, any additional comments before merging?

**Comment by komidore64** [comment]
:expressionless: Looks like another rebase is needed.

No additional comments from me.

**Comment by ehelms** [comment]
Now that we have a test setup, what do you think about adding some for these updates?

**Comment by adamruzicka** [comment]
@ehelms Added some tests which should cover the updates

File: lib/hammer_cli_katello/id_resolver.rb [hunk 1]
```diff
@@ -61,6 +62,18 @@ def repository_id(options)
...
+    def repository_ids(options)
+      return options['option_repository_ids'] unless options['option_repository_ids'].nil?
+
+      key_names = HammerCLI.option_accessor_name 'names'
+      key_product_id = HammerCLI.option_accessor_name 'product_id'
+      options[key_product_id] ||= product_id(scoped_options 'product', options)
+      find_resources(:repositories, options)
+        .select { |repo| options[key_names].include? repo['name'] }.map { |repo| repo['id'] }
+      # ^ This could be removed if Katello's repositories controller allowed to filter
+      #   repositories index by repository names
```
Comment (ehelms) [inline lib/hammer_cli_katello/id_resolver.rb]: I tend towards not including this comment, this is a general issue with how tenancy works in Katello and I don't want someone reading this thinking it is a simple change to the repositories controller. Right now filtering by name can occur via scoped_search or if we combined name + product to scope things properly.

**Comment by ehelms** [comment]
Thanks for adding tests! @tstrachota or @mbacovsky would you mind taking a look at the tests briefly? Since we haven't had tests in here before, I am still getting up to speed on what a good test within hammer should look like.

File: lib/hammer_cli_katello/id_resolver.rb (hunk 1 above)
Comment (adamruzicka) [inline lib/hammer_cli_katello/id_resolver.rb]: I see, removed

File: test/functional/content-view/create_test.rb [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    api_expects(:content_views, :create, 'Create content view') do |par|
+      par['name'] == name &&
+        par['repository_ids'] == ids &&
+        par['organization_id'] == org_id
+    end
+
+    expected_result = success_result("Content view created\n")
+    result = run_cmd(@cmd + @base_params + params)
+    sanitize.call(result)
```
Comment (tstrachota) [inline test/functional/content-view/create_test.rb]: If the warning is expected behavior, I'd prefer changing the stderr expectation over clearing the result. 
```ruby
expected_result = success_result("Content view created\n")
expected_result.expected_err = 'Warning: ...'
result = run_cmd(@cmd + @base_params + params)
assert_cmd(expected_result, result)
```

**Comment by tstrachota** [comment]
@ehelms @adamruzicka One note in the tests. Otherwise it looks good!

File: test/functional/content-view/create_test.rb (hunk 2 above)
Comment (adamruzicka) [inline test/functional/content-view/create_test.rb]: Created issues for the deprecation warning problem[1] and test refactoring[2]

[1] - http://projects.theforeman.org/issues/14289
[2] - http://projects.theforeman.org/issues/14290

Comment (tstrachota) [inline test/functional/content-view/create_test.rb]: Just for the record:
After a further investigation it turned out the printed warning is a bug. The warning should be displayed only when users really pass `--id`. Currently it occurs every time the content-view command is loaded.

Comment (tstrachota) [inline test/functional/content-view/create_test.rb]: One more idea:
```ruby
require 'hammer_cli_katello/content_view'
```
is probably cleaner workaround than deleting stderr.

Comment (adamruzicka) [inline test/functional/content-view/create_test.rb]: Incorporated that into the latest commit, works like a charm. Thanks!

**Comment by adamruzicka** [comment]
@bbuckingham @ehelms @komidore64 any more comments on this?

**Comment by komidore64** [comment]
None from me.

**Comment by ehelms** [comment]
Nothing from me -- looks like it needs a rebase

**Comment by adamruzicka** [comment]
Rebased and the tests are green

**Comment by ehelms** [comment]
ACK - thanks @adamruzicka !

**Pull request merged**